### PR TITLE
Add initial version of the C++ styleguide

### PR DESCRIPTION
**Pull request by zlobober**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: styleguide/cpp.md [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+
+<!------------------------------------------------------------------------------------>
+
+### Logging
+
+Log messages follow quite strict rules. This results in an extreme convenience for debugging and searching for relevant log messages whenever you get used to it.
+
+Logging is done only using the `YT_LOG_XXX` macros. 
+
+Log messages are usually in past simple, present continuous or do not contain a verb at all. Log messages are not completed English sentences, they do not finish with dot. In case when multiple "sentences" must be present within a single message, separate them with a semicolon.
```
Comment (achulkov2) [inline styleguide/cpp.md:515]: dot -> period

File: styleguide/cpp.md [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+
+We use a quote-style include directives for the headers located in the same directory as the including file. We use angle-bracket-style include directives for the rest of included headers, including system ones.
+
+Included headers are grouped by the base directory they are located in. Groups are separated by a blank line. Within a group, headers are sorted alphabetically.
+
+We order groups from the most specific to the most general. Motivation is that we would like to reduce the number of situations when the latter included header already includes the former one as it may hide some dependency issues. 
+
+We do not care about including everything which is used by the current file.
+
+For .cpp files, the corresponding header is always the first included header and is separated by a blanck line.
```
Comment (Krock21rus) [inline styleguide/cpp.md:205]: blanck -> blank

File: styleguide/cpp.md [hunk 3]
```diff
@@ -0,0 +1,873 @@
...
+// Ok.
+YT_LOG_DEBUG("Tables fetched (TableCount: %v)", tableCount);
+// Not ok.
+YT_LOG_DEBUG("Fetched %v tables", tableCount);
+YT_LOG_DEBUG("Tables fetched (tableCount: %v)", tableCount);
+```
+
+### Errors
+
+All error and exception text must be written as if they were dedicated to an end user. They must also be written in a gramatically correct English; typicall they also do not contain dots, having semicolons instead.
```
Comment (Krock21rus) [inline styleguide/cpp.md:554]: typicall -> typically

Comment (zlobober) [reply styleguide/cpp.md:554]: Spell checker was something I forgot for sure.

**Comment by zlobober** [comment]
@chizhonkova, what's ETA on merging this?

**Comment by chizhonkova** [comment]
I'll merge it on this week

**Review by chizhonkova — APPROVED** [review]

**Pull request closed without merging**